lib/trino-hive-formats/src/main/java/io/trino/hive/formats/avro/model/AvroLogicalType.java

```diff
@@ -0,0 +1,112 @@
+        AvroLogicalType.DateLogicalType,
+        AvroLogicalType.TimeMillisLogicalType,
+        AvroLogicalType.TimeMicrosLogicalType,
+        AvroLogicalType.StringUUIDLogicalType
+{
+    static AvroLogicalType fromAvroLogicalType(LogicalType logicalType, Schema schema)
+    {
+        switch (logicalType.getName()) {
+            case TIMESTAMP_MILLIS -> {
+                if (schema.getType() == Schema.Type.LONG) {
+                    return new TimestampMillisLogicalType();
+                }
+            }
+            case TIMESTAMP_MICROS -> {
+                if (schema.getType() == Schema.Type.LONG) {
+                    return new TimestampMicrosLogicalType();
+                }
+            }
+            case DECIMAL -> {
+                LogicalTypes.Decimal decimal = (LogicalTypes.Decimal) logicalType;
+                switch (schema.getType()) {
+                    case BYTES -> {
+                        return new BytesDecimalLogicalType(decimal.getPrecision(), decimal.getScale());
+                    }
+                    case FIXED -> {
+                        return new FixedDecimalLogicalType(decimal.getPrecision(), decimal.getScale(), schema.getFixedSize());
+                    }
+                    default -> throw new IllegalArgumentException("Unsupported Logical Type %s and schema pairing %s".formatted(logicalType.getName(), schema));
+                }
+            }
+            case DATE -> {
+                if (schema.getType() == Schema.Type.INT) {
+                    return new DateLogicalType();
+                }
+            }
+            case TIME_MILLIS -> {
+                if (schema.getType() == Schema.Type.INT) {
+                    return new TimeMillisLogicalType();
+                }
+            }
+            case TIME_MICROS -> {
+                if (schema.getType() == Schema.Type.LONG) {
+                    return new TimeMicrosLogicalType();
+                }
+            }
+            case UUID -> {
+                if (schema.getType() == Schema.Type.STRING) {
+                    return new StringUUIDLogicalType();
+                }
+            }
+        }
+        throw new IllegalArgumentException("Unsupported Logical Type %s and schema pairing %s".formatted(logicalType.getName(), schema));
+    }
+
+    record TimestampMillisLogicalType()
+            implements AvroLogicalType {}
+
+    record TimestampMicrosLogicalType()
+            implements AvroLogicalType {}
+
+    record BytesDecimalLogicalType(int precision, int scale)
+            implements AvroLogicalType {}
+
+    record FixedDecimalLogicalType(int precision, int scale, int fixedSize)
+            implements AvroLogicalType {}
+
+    record DateLogicalType()
+            implements AvroLogicalType {}
+
+    record TimeMillisLogicalType()
+            implements AvroLogicalType {}
+
+    record TimeMicrosLogicalType()
+            implements AvroLogicalType {}
+
+    final class StringUUIDLogicalType
```

Comment:
Why is this one a class and the rest are records?

Reply:
forgot to convert